src/documents/CoreAPI.md

```diff
@@ -0,0 +1,675 @@
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied Infusion component has been [destroyed](ComponentLifecycle.md).
+
+### fluid.hasGrade(options, gradeName)
+
+* `options {Object}`
+* `gradeName {String}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied options ([defaults](#fluid-defaults-gradename-options-)) are the top-level options of a component which has the supplied grade. These must be options which
+have already been through options merging, since the test is a simpleminded one that simply checks whether the `gradeNames` entry in the options contains the supplied `gradeName`.
+
+### fluid.componentHasGrade(component, gradeName)
+
+* `component {Component}`
+* `gradeName {String}`
+* Returns: `{Boolean}`
+
+Returns `true` if the supplied component has the supplied `gradeName` as a parent grade. This is a rapid check which uses the framework's internal data structures for the lookup.
+
+
+## Basic implementation atoms
+
+### fluid.identity(arg)
+
+* `arg {Any}`
+* Returns: `{Any}`
+
+### fluid.notImplemented()
+
+A function which raises a failure if executed. This function name can also be used as a placeholder in [Invokers](Invokers.md) and
+namespaced [Listener](InfusionEventSystem.md) to create abstract (noninstantiable) [Component Grades](ComponentGrades.md).
+
+### fluid.each(source, func)
+
+* `source {Array|Object}` A container to be iterated over (may be null)
+* `func {Function} (element {Any}, index {Number|String}) → Any`. This function will be invoked once on each member of `source` and will receive the member and its key (either an integer for an array source,
+or a string for an object source) 
+* Returns: none
+
+Iterates over a supplied array or hash, invoking a function once for each member. Similar to [`jQuery.each`](http://api.jquery.com/jquery.each/) only the
+arguments to `func` are the right way round and the function does not explode on nonvalues. `fluid.each` on `null` or `undefined` is a no-op.
+
+### fluid.invokeLater(func) 
+
+* `func {Function}` A function accepting zero arguments which is to be invoked later
+* Returns: none
+
+A standard utility to schedule the invocation of a function after the current function call stack returns. On browsers this defaults to `setTimeout(func, 1)` but in
+other environments can be customised - e.g. to `process.nextTick` in node.js. In future, this could be optimised in the browser to not dispatch into the event queue.
+
+### fluid.allocateGuid()
+
+Allocate a string value that will be unique within this Infusion instance (frame or process), and
+globally unique with high probability (50% chance of collision after a million trials). This has the structure `prefix-id` where
+`prefix` is the value of `fluid.fluidInstance` identifying this Infusion instance, and `id` is a increasing integer initialised at
+1 when this infusion instance starts up.
+
+* Returns: `{String}` A string of about a dozen characters that has made moderate efforts to be globally unique.
```

Comment:
For consistency, the description of allocateGuid() should to after the description of the return value.